### DISCUSSION TITLE
Would it be possible to install Brew without root access on Linux ?

### DISCUSSION THREAD

**Discussion by HGStyle** [Homebrew/discussions#5236] [Linux]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
[DISCUSSION'S AUTHOR NOTE: THIS IS THE OUTPUT FOR MY MACHINE ON WSL, NOT ON MY SERVER!]
HOMEBREW_VERSION: 4.2.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 92a4311868322188478d7a90511ec0e8e6b0d7df
Last commit: 3 days ago
Core tap JSON: 27 Mar 07:46 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dodeca-core 64-bit unknown_0x6_0x9a
Clang: 14.0.0
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.146.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.4 LTS
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 13.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
[DISCUSSION'S AUTHOR NOTE: THIS IS THE OUTPUT FOR MY MACHINE ON WSL, NOT ON MY SERVER!]
Your system is ready to brew.
```

### Description of issue

I have a server where I have shell access, but no root access. I got build tools such as GCC, GNU BinUtils (includes the assembler), GNU Make, and CMake if I remember correctly. I have made a little "fake prefix" in $HOME/usr which I use when building software with `--prefix`. I can build many softwares manually but binaries are mostly not working. The CPU is a basic x86_64 and building from source is not a real problem when building one software since I have a lot of RAM on it. The problem with binaries are libraries: when the binary is not fully statically linked, it's not working because lots of libraries are not installed. So I got some questions about Homebrew:

1. Would it be possible to install Homebrew to annother prefix (because I have no write rights in /home) like to $HOME/.linuxbrew or something like that ?
2. Is it possible to install it without using SUDO or similar commands ?
3. Do Homebrew and its "precompiled softwares" (I heard it's called "bottles") would work ? (NOTE: I have all the [Linux requirements](https://docs.brew.sh/Homebrew-on-Linux#requirements) met: GCC 8.5, GLibC 2.28 and Linux 4.18).

Thanks you in advance.

**Comment by SMillerDev** [comment]
I think https://docs.brew.sh/Homebrew-on-Linux answers a lot of your questions about prefixes

**Reply by HGStyle** [reply]
Thanks, I just read it.
>"You shouldn’t install outside the default, supported, best prefix. Many things will need to be built from source outside the default prefix. [...] The main reason Homebrew just works is because we use bottles (binary packages) and most of these require using the default prefix."

Why are bottles not working when using annother prefix ? Because I would really like to install it but as I said in the post, I have no write rights in the /home directories and the admins are not gonna create it, I allready asked them for a software that was not present (Java) and they never installed it or told me how to install it.

**Reply by SMillerDev** [reply]
> Why are bottles not working when using annother prefix ?

Because a lot of software only works with the settings you compile it with. If you tell something it'll run in `/home/linuxbrew/.linuxbrew` on compilation and then run it in another path, unexpected things might happen.

You could try `--force-bottle` but no guarantees it'll work

**Comment by xiaoran007** [comment]
You can just export export HOMEBREW_RELOCATE_BUILD_PREFIX to you homebrew prefix, and then you can download pre-build binary package rather than rebuild them.

**Reply by HGStyle** [reply]
Ooh, I'll look into that, thanks!
